security/pkg/nodeagent/cache/secretcache_test.go

```diff
@@ -1175,3 +1175,31 @@ func TestWorkloadAgentGenerateSecretFromFile(t *testing.T) {
 		t.Errorf("Unused secrets failed to be evicted from cache")
 	}
 }
+
+func TestShouldRotate(t *testing.T) {
+	now := time.Now()
+	secret := &model.SecretItem{
+		ResourceName: "Test-resource",
+		ExpireTime:   now.Add(time.Hour),
+		CreatedTime:  now.Add(-time.Hour),
+	}
+
+	options := &Options{SecretRotationGracePeriodRatio: 0.4}
+	sc := &SecretCache{configOptions: *options}
+	if sc.shouldRotate(secret) {
+		t.Errorf("Should not rotate: not in grace period.")
+	}
+
+	options.SecretRotationGracePeriodRatio = 0.6
+	sc = &SecretCache{configOptions: *options}
+	if !sc.shouldRotate(secret) {
+		t.Errorf("Should rotate: in grace period.")
+	}
+
+	secret.ExpireTime = now.Add(-time.Minute)
+	options.SecretRotationGracePeriodRatio = 0
+	sc = &SecretCache{configOptions: *options}
+	if !sc.shouldRotate(secret) {
+		t.Errorf("Should not rotate: passed the expiration.")
```

Comment:
this seems like as should rotate if it is expired?

Reply:
Sorry, typo...